cpp/include/raft/comms/comms_helper.hpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <iostream>
+#include <raft/comms/comms.hpp>
+#include <raft/handle.hpp>
+#include <raft/mr/device/buffer.hpp>
+
+namespace raft {
+namespace comms {
+
+void build_comms_nccl_only(handle_t *handle, ncclComm_t comm, int size,
```

Comment:
The reason for these factory-like methods is to abstract the injection of the conms on the handle from Python, rather than having to expose these methods on the handle. We could do that, but it seems cleaner to have cython call these functions. Otherwise, I do agree, but I wouldn’t expect someone in the c++ layer to ever be calling  these functions directly.

Reply:
Got it, makes sense if this is mainly for cython interface.

Reply:
I could rename the file to something like `cython_helper.hpp`, but I'd prefer that we not limit its potential use. What do you think about a comment in the code that specifies these functions help enable the cython layer?

Reply:
Yes, that comment will be very helpful. Comments are generally very helpful if we make some non-obvious design choices/work-arounds due to practical restrictions.